src/engine/test/source/hlp2/basic_test.cpp

```diff
@@ -301,3 +301,46 @@ TEST(HLP2, TextParser)
         runTest(t, hlp::getTextParser);
     }
 }
+
+TEST(HLP2, AlphanumericParser)
+{
+    ASSERT_THROW(hlp::getAlphanumericParser({}, {}, {"arg"}), std::runtime_error);
+
+    auto fn = [](std::string in) -> json::Json
+    {
+        json::Json doc;
+        doc.setString(in);
+        return doc;
+    };
+    std::vector<TestCase> testCases {
+        TestCase {"id98A",
+                true,
+                {},
+                Options(),
+                fn("id98A"),
+                5},
+        TestCase {"id98A:69",
+                false,
+                {},
+                Options(),
+                fn("id98A:69"),
+                0},
+        TestCase {"idARGbyu",
+                true,
+                {},
+                Options(),
+                fn("idARGbyu"),
+                8},
+        TestCase {"0123456789",
+                true,
+                {},
+                Options(),
+                fn("0123456789"),
+                10}
```

Comment:
Please, add some more cases including characters like ' ', '-', '_' and any other that may be captured as "Alphanumerical